substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/debugentry/DebugInfoBase.java

```diff
@@ -657,7 +717,7 @@ public boolean isHubClassEntry(ClassEntry classEntry) {
     }
 
     public int classLayoutAbbrevCode(ClassEntry classEntry) {
-        if (useHeapBase & isHubClassEntry(classEntry)) {
+        if (!useHeapBase & isHubClassEntry(classEntry)) {
```

Comment:
Just checking: this change is intended, right? Does it need a comment?

Reply:
Yes, it is intentional. The old test was wrong. However, it was also never called -- the relevant logic is implemented directly in the code that writes the class layouts where its significance is [explained](https://github.com/adinn/graal/blob/8428e0e3bc9df9a1bb66ee03e1536b7cecac3c1d/substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/elf/dwarf/DwarfInfoSectionImpl.java#L431)

The check is only relevant to the DWARF generator which means it ought not to be defined in generic (debugentry) code and it is better explained at the point of use. So, I'll simply remove this method.

Reply:
Done